google/cloud/storage/object_stream.h

```diff
@@ -343,6 +343,15 @@ class ObjectWriteStream : public std::basic_ostream<char> {
    */
   void Suspend() &&;
 
+  /**
+   * Returns the status of partial errors.
+   *
+   * Application may write multiple times before closing the stream, this
+   * function gives the capability to find out status even before stream
+   * closure.
+   */
+  Status const& last_status() const { return buf_->last_status(); }
```

Comment:
Since this is part of the public API, `last_status()` may be confused to have a similar role to `ObjectReadStream`'s `status()`. Maybe this should document that `last_status()` has a relationship to `last_response` and signals different errors not already covered by `metadata()`.

Reply:
Alternatively, if `last_status()` overlaps with `metadata().status()` maybe this can just be renamed to `status()`?

Reply:
AFAIK, `metadata()` and `last_response` shows the final status where as `last_status()` can be used to find intermediate `UploadChunk` status or partial errors. Let me update comments for `last_status()`

Reply:
Thanks, makes sense. What do you think about renaming it to `latest_status` or something along those lines? But I'll defer to @coryan for that.

Nit: s/then/than/